woocommerce/woocommerce
# Validate orders separately since it not always a CPT.

### All Submissions:






Targets #36650 

### Changes proposed in this Pull Request:



In the webhook code, we are still relying on `get_post_status` method to get status of an order, which does not works with HPOS. This PR splits that logic to use `$order->get_status` when the webhook is for order instead.

Closes #36686 






### How to test the changes in this Pull Request:




Check that the unit test for `WC_Webhook_Test` are passing now using the command `./vendor/bin/phpunit -c phpunit.xml --filter=WC_Webhook_Test`. These tests were earlier failing in #36650.


Alternatively, this can be tested by going creating and testing a webhook from UI path like so:

1. Go to WooCommerce > Settings > Advanced > Webhooks and add a new webhook which targets `order.created` event. Add any dummy URL that can't be reached, such as w.test (we will inspect the log to see if webhook delivery was attempted). Save the webhook, and ignore the invalid URL error for now.
2. Add this filter somewhere in code, for example, in functions.php or via the Code Snippets plugin: `apply_filters( 'woocommerce_webhook_deliver_async', '__return_false' );`
3. Create a new order, either via checkout, or by going to WooCommerce > Orders > Create New. This should trigger the webhook.
4. To verify that webhook delivery was attempted, go to WooCommerce > Status > Logs and check that there will be a new log file for failed webhook delivery. Since this verified that webhook delivery was attempted, this is testing enough. If you have used a correct and observable URL in step 1, you will see that webhook was indeed delivered to the provided URL.
 



### Other information:




### FOR PR REVIEWER ONLY:
